### PR TITLE
Update UG/DG

### LINKED ISSUE #145: Add an architecture overview diagram

### LINKED ISSUE #157: [PE-D][Tester B] Missing Non-Functional Requirements in DG
Non-Functional Requirements is missing in DG. Adding these can be helpful to allow the reader to understand more about the design considerations when it comes to engineering the product.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: DaDevChia/ped#3

### LINKED ISSUE #167: [PE-D][Tester B] UG: Incomplete FAQ
Curreently the FAQ only has 1 question and there is no answer provided to the question. Would a more comprehensive FAQ be better to help users identify common issues?

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: DaDevChia/ped#8

### PR DESCRIPTION
**Pull request by ryan1604**
Fixes #145 
Fixes #157 
Fixes #167

### CODE REVIEW

File: docs/images/ArchitectureDiagram.png
Comment (NeoMinWei) [inline docs/images/ArchitectureDiagram.png:1]: storage quite hard to see

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -336,7 +353,9 @@ Example: `budget view`
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+* Should work on main OS (Windows, Linux, Mac) that has Java 11 installed.
+* This app is meant for a single user.
+* This app targeted towards users with an above-average typing speed.
```
Comment (NeoMinWei) [inline docs/DeveloperGuide.md]: is targeted*

**Review by NeoMinWei — APPROVED** [review]
LGTM

File: src/main/java/seedu/financialplanner/utils/Ui.java [hunk 2]
```diff
@@ -66,9 +66,9 @@ public String input() {
     }
 
     public void printWatchListHeader() {
-        showMessage("Symbol");
+        System.out.print(CYAN + "Symbol" + RESET);
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/utils/Ui.java:69]: Thanks!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 * [Acknowledgements](#acknowledgements)
 * [Design & implementation](#design--implementation)
+  * [Architecture Diagram](#architecture-diagram)
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:7]: Nice however the colours are a little too bright

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -70,7 +71,23 @@
...
+
+`Main` consists of the `FinancialPlanner` class, and is in charge of starting up and shutting down the app.
+
+The program consists of five main components:
+
+* `Ui`: User interactions.
+* `Parser`: Parse user inputs.
+* `Command`: The command executor.
+* `Model`: Holds the data of the program in memory.
+* `Storage`: Reads data from, and writes data to the hard disk.
```
Comment (wwweert123) [inline docs/DeveloperGuide.md:90]: Do you want to add our important classes such as cashflowlist?

File: docs/team/ryan1604.md [hunk 5]
```diff
@@ -32,14 +33,22 @@ you a one-stop interface to access a plethora of features to manage your finance
   * Class diagram
   * Design considerations
 * Budget feature
+  * Implementation
+  * Sequence diagrams
+* Non-functional requirements
+* Architecture Diagram
```
Comment (wwweert123) [inline docs/team/ryan1604.md:39]: Thankyou!

**Review by wwweert123 — APPROVED** [review]

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ryan1604) [reply docs/DeveloperGuide.md:90]: Thats all under model:)

**Pull request merged**